src/kernel/parser/grammar.y

```diff
@@ -5677,19 +5679,14 @@ static int yylex(rb_parse_state* ps)
         return '~';
 
       case '(':
-        ps->command_start = TRUE;
-        if (IS_EXPR_BEG_or_MID(lex_state)) {
+        if (IS_BEG()) {
             c = tLPAREN;
         }
         else if (space_seen) {
-            if (lex_state == EXPR_CMDARG) {
-                c = tLPAREN_ARG;
-            }
-            else if (lex_state == EXPR_ARG) {
-                rb_warning(ps, "don't put space before argument parentheses");
-                c = '(';
-            }
+          c = tLPAREN_ARG;
         }
+
+        // TODO: paren_nest++;
```

Comment:
What are the issues with this todo pending? Do you have any code that doesn't parse because of this in particular?